### PR TITLE
Improves "Could not reach Firestore backend." log message.

### PR DESCRIPTION
**Pull request by mikelehen**
Improved message:

> Could not reach Cloud Firestore backend. Connection failed 2 times. Most recent error: FirebaseError: [code=unavailable]: 14 UNAVAILABLE: Connect Failed
> This is purely an informational message and typically indicates that your device does not have a healthy Internet connection at the moment. The client will operate in offline mode until it is able to successfully connect to the backend.
> 

Or:

> Could not reach Cloud Firestore backend. Backend didn't respond within 10 seconds.
> This is purely an informational message and typically indicates that your device does not have a healthy Internet connection at the moment. The client will operate in offline mode until it is able to successfully connect to the backend.

### CODE REVIEW

File: packages/firestore/src/remote/online_state_tracker.ts [hunk 1]
```diff
@@ -166,10 +170,18 @@ export class OnlineStateTracker {
     }
   }
 
-  private logClientOfflineWarningIfNecessary(): void {
+  private logClientOfflineWarningIfNecessary(details: string): void {
+    const message =
+      `Could not reach Cloud Firestore backend. ${details}\n` +
+      `This is purely an informational message and typically indicates ` +
```
Comment (schmidt-sebastian) [inline packages/firestore/src/remote/online_state_tracker.ts]: I would remove the first part of this sentence. Just saying "This typically indicates that your device ..." should make it obvious where the problem is.

File: packages/firestore/src/remote/remote_store.ts [hunk 2]
```diff
@@ -315,16 +315,17 @@ export class RemoteStore {
...
+  private async onWatchStreamClose(error?: FirestoreError): Promise<void> {
     assert(
       this.isNetworkEnabled(),
       'onWatchStreamClose() should only be called when the network is enabled'
     );
 
     this.cleanUpWatchStreamState();
-    this.onlineStateTracker.handleWatchStreamFailure();
+
+    if (error) {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/remote/remote_store.ts]: I am tempted to make this part of the `this.shouldStartWatchStream()` if-check. That way we only change the state to `Unknown` if we actually care (and nicely works around the problem that the comment in line 330 seems to apply more to line 326 than 331).

File: packages/firestore/src/remote/online_state_tracker.ts (hunk 1 above)
Comment (mikelehen) [reply packages/firestore/src/remote/online_state_tracker.ts]: Good call, done.

File: packages/firestore/src/remote/remote_store.ts (hunk 2 above)
Comment (mikelehen) [reply packages/firestore/src/remote/remote_store.ts]: done (and tweaked comments)

**Comment by schmidt-sebastian** [comment]
Thanks. Changes look good.

**Pull request merged**